File: Data Structures & Algorithms/concatenation-of-array/submission-0.rs
impl Solution {
    pub fn get_concatenation(nums: Vec<i32>) -> Vec<i32> {
        if nums.len() == 0{
            return vec![];
        }
        let n = nums.len();
        let mut ans = vec![0; n * 2];
        for (i, a) in nums.iter().enumerate() {
            ans[i] = a.clone();
            ans[i + n] = a.clone();
        }
        return ans;
    }
}

File: Data Structures & Algorithms/first-missing-positive/submission-0.cpp
class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        int ans = 1;
        for(int i=0; i<nums.size(); i++){
            if(nums[i] < 0){
                nums[i] = 0;
            }
        }
        for(int i=0; i<nums.size(); i++){
            int val = abs(nums[i]);
            if(1 <= val && val<= nums.size()){
                if (nums[val - 1] > 0){
                    nums[val-1] *= -1;
                }else if(nums[val-1] == 0){
                    nums[val-1] = -(nums.size() + 1);
                }
            }
        }
        for(int i=0; i<nums.size(); i++){
            if(nums[i] >= 0 )
                return i+1;
        }
        return nums.size()+1;
    }
};